game-app/game-core/src/main/java/games/strategy/triplea/attachments/RulesAttachment.java

```diff
@@ -507,6 +510,39 @@ private void resetIsAI() {
     isAI = null;
   }
 
+  @VisibleForTesting
+  public void setHasResource(final String value) throws GameParseException {
+    final String[] s = splitOnColon(value);
```

Comment:
Follow [G1. Choose variable names that provide useful information](https://openjdk.org/projects/amber/guides/lvti-style-guide#G1)